server/pbench/bin/index-pbench

```diff
@@ -1011,6 +1149,29 @@ class ToolData(object):
             datafiles.append(datafile)
         return datafiles
 
+    @staticmethod
+    def get_stdout_files(handler, iteration, sample, host, tool, toolgroup, ptb):
+        """
+        Fetch the stdout file for this tool returning a list of dicts
+        containing metadata.
+        """
+        stdout_file = "{0}-stdout.txt".format(tool)
+        path = os.path.join(iteration, sample, "tools-%s" % (toolgroup,), host, tool, stdout_file)
+        paths = [x for x in ptb.tb.getnames() if x.find(path) >= 0 and ptb.tb.getmember(x).isfile()]
+        datafiles = []
+        for p in paths:
+            fname = os.path.basename(p)
+            for rec in handler['patterns']:
+                if rec['pattern'].match(fname):
+                    handler_rec = rec
+                    break
+
+            assert handler_rec is not None
```

Comment:
Instead of an `assert` it seems like you can use the `for/else` construct here to where the `else` clause is executed when the for loop goes through all its iterations [1].

[1] https://docs.python.org/2/reference/compound_stmts.html#the-for-statement

Reply:
Yeah I copied that from `get_csv_files()` and thought it was a good idea.